evennia/utils/opclasses.py

```diff
@@ -0,0 +1,319 @@
+
+        Returns:
+            Boolean: Whether loading was successful.
+        """
+        if self.save_data is not None:
+            try:
+                self.value_storage = self.valid_save(self.save_data)
+                self.loaded = True
+                return True
+            except Exception as e:
+                print(e)  # need some kind of error message here!
+        return False
+
+    def customized(self):
+        return self.value_storage != self.default_value
+
+    def valid_save(self, save_data):
+        """
+        Perform sanity-checking on the save data. This isn't the same as Validators, as Validators deal with
+        user input. save data might be a timedelta or a list or some other object. isinstance() is probably
+        very useful here.
+
+        Args:
+            save_data: The data to check.
+
+        Returns:
+            Arbitrary: Whatever the Option needs to track, like a string or a datetime. Not the same as what
+                users are SHOWN.
+        """
+        return save_data
+
+    def clear(self):
+        """
+        Resets this Option to default settings.
+
+        Returns:
+            self. Why?
+        """
+        self.value_storage = None
+        self.loaded = False
+        return self
+
+    @property
+    def default(self):
+        return self.default_value
+
+    @property
+    def value(self):
+        if not self.loaded and self.save_data is not None:
+            self.load()
+        if self.loaded:
+            return self.value_storage
+        else:
+            return self.default
+
+    def validate(self, value, account):
+        """
+        Validate user input, which is presumed to be a string.
+
+        Args:
+            value (str): User input.
+            account (AccountDB): The Account that is performing the validation. This is necessary because of
+                other settings which may affect the check, such as an Account's timezone affecting how their
+                datetime entries are processed.
+
+        Returns:
+            The results of a Validator call. Might be any kind of python object.
+        """
+        return self.do_validate(value, account)
+
+    def do_validate(self, value, account):
+        """
+        Second layer of abstraction on validation due to design choices.
+
+        Args:
+            value:
+            account:
+
+        Returns:
+
+        """
+        return self.valid[self.valid_type](value, thing_name=self.key, account=account)
+
+    def set(self, value, account):
+        """
+        Takes user input, presumed to be a string, and changes the value if it is a valid input.
+
+        Args:
+            value:
+            account:
+
+        Returns:
+
+        """
+        final_value = self.validate(value, account)
+        self.value_storage = final_value
+        self.loaded = True
+        self.save()
+        return self.display()
+
+    def display(self):
```

Comment:
Why not accept `**kwargs` here so the display can be influenced by whomever looking? Then you don't need to keep passing that `account` argument around elsewhere.